### PR TITLE
Bug/aggregate bids

### LINKED ISSUE #246: Aggregate Bid Data
## Detailed Description
The subgraph pushes out each individual bid and assumes that each bid is able to be claimed seperately. The data should be aggregated and updated on the `SelfBidList` component such that the user will realize that they only need to claim their tokens once. 

## Context
Important for user clarity. 

## Possible Implementation
reduce the array and update the component for each bid.

## Your Environment

- Version used:
- Environment name and version (e.g. Chrome 39, node.js 5.4):
- Operating System and version (desktop or mobile):
- Link to your project:

### LINKED ISSUE #269: Hide unneeded buttons on closed sales
- Over Soft-Cap: Only show "Claim Tokens" 
- Under Soft-Cap: Only show "Withdraw" 
- Change from "Withdraw failed bids" to "Withdraw" in fixed price context

### PR DESCRIPTION
**Pull request by hhamud**
---
name: Pull Request
about: Create a pull request for this project

---

## Description
* Closes #246 
* Closes #269 

## How Has This Been Tested?
Tested Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by rossneilson — APPROVED** [review]
Looks good to me.
Nice to see it aggregate bids.
If the information is there then it could be good to have an option to show more info of the individual purchases in a drop down menu.

Although I did find another issue, it seems once funds are withdrawn they are not removed from the UI or marked as withdrawn.

File: src/views/Sale/components/SelfBidList/index.tsx [hunk 1]
```diff
@@ -123,6 +123,7 @@ export function SelfBidList({ sale, clearingPrice, bids, isFixed }: SelfBidListP
     ? formatBigInt(clearingPrice.tokenIn, sale.tokenIn.decimals) /
       formatBigInt(clearingPrice.tokenOut, sale.tokenOut.decimals)
     : 0
+  console.log(bids)
```
Comment (adamazad) [inline src/views/Sale/components/SelfBidList/index.tsx:126]: Remove logs

Comment (adamazad) [inline src/views/Sale/components/SelfBidList/index.tsx:126]: Remove logs

**Review by adamazad — CHANGES_REQUESTED** [review]

File: src/hooks/useBids.ts [hunk 2]
```diff
@@ -23,6 +24,7 @@ interface UseBidsReturn {
   error: Error | null
   bids: SaleBid[]
   totalBids: SaleBid[]
+  totalPurchased(bids: SaleBid[]): Pick<FixedPriceSalePurchase, 'buyer' | 'amount'>[]
 }
 
 export function useBids(saleId: string, saleType: SaleType): UseBidsReturn {
```
[the remark this answers is not in the document]
Comment (hhamud) [reply src/hooks/useBids.ts:30]: I was using the mesa sdk originally. I can switch it out for those hook's you've made later on

File: src/hooks/useTokenClaim.ts [hunk 3]
```diff
@@ -16,15 +16,11 @@ interface useTokenClaimReturns {
...
   withdrawTokens: (saleId: string) => void
-  withdrawError: MetaMaskError | undefined
-  claimWithdraw: Claim
 }
 
 export function useTokenClaim(): useTokenClaimReturns {
   const [claim, setClaim] = useState<Claim>('unclaimed')
-  const [claimWithdraw, setClaimWithdraw] = useState<Claim>('unclaimed')
   const [error, setError] = useState<MetaMaskError>()
-  const [withdrawError, setWithdrawError] = useState<MetaMaskError>()
```
[the remark this answers is not in the document]
Comment (hhamud) [reply src/hooks/useTokenClaim.ts:27]: I will make a separate file for the error types

**Pull request merged**